nixos/modules/config/networking.nix

```diff
@@ -192,6 +198,33 @@ in
 
   config = {
 
+    assertions = [{
+      assertion = localhostMapped4;
+      message = "`networking.hosts` doesn't map `127.0.0.1` to `localhost`";
+    } {
+      assertion = !cfg.enableIPv6 || localhostMapped6;
+      message = "`networking.hosts` doesn't map `::1` to `localhost`";
+    } {
+      assertion = !localhostMultiple;
+      message = ''
+        `networking.hosts` maps a non-loopback address to `localhost`. Generated
+        `/etc/hosts` might make `localhost` resolve to that non-loopback address instead
+        of `127.0.0.1` and `::1` which would break some applications. Please use
```

Comment:
This message should be simplified. It is also not very correct, because one may map "127.0.0.2" to localhost and it would be perfectly correct mapping